feat: add `ParentIsA` to force parent child relations (#1566)
flame-engine/flame: doc/flame/components.md

that they will appear in the children list in the same order as they were
scheduled for addition.

> #### Ensuring a component has a given parent

Author: I did that as a fourth as I didnt find a good place to put this, and this chapter talks about children order and parent relations so I thought this might be a nice fit.

> #### Ensuring a component has a given parent

Reviewer: I think this would be fine as a third level header too, if you look at the section after this one it is also third level and I think they fit on the same level.

When a component requires to be added to a specific parent type the 
`HasParent` mixin can be used to enforce a strongly typed parent.

Example:

```dart
class MyComponent extends Component with HasParent<MyParentComponent> {

>   Future<void> onLoad() async {

Reviewer 2: Add `@override`

    // parent is of type MyParentComponent
    print(parent.myValue);
  }
}
```

> If `MyComponent` is added to a component that is not `MyParentComponent` 
> an assertion error will be thrown stating such.

Reviewer 2: ```suggestion
If you try to add `MyComponent` to a parent that is not `MyParentComponent`, 
an assertion error will be thrown.
```
